test/test_diagnostics.jl

```diff
@@ -39,42 +37,121 @@ function diagnostic_windowed_spatial_average(arch, FT)
     return u_mean(model)[1] == 7
 end
 
-function advective_cfl_diagnostic_is_correct_on_vertically_stretched_grid(arch, FT)
-    model = TestModel_VerticallyStrectedRectGrid(arch, FT)
+function diffusive_cfl_diagnostic_is_correct(arch, FT)
+    Δt = FT(1.3e-6)
+    Δx = FT(0.5)
+    ν = FT(1.2)
+    CFL_by_hand = Δt * ν / Δx^2
+
+    model = TestModel_RegularRectGrid(arch, FT, ν, Δx)
+    cfl = DiffusiveCFL(FT(Δt))
+
+    return cfl(model) ≈ CFL_by_hand
+end
+
+function advective_cfl_diagnostic_is_correct_on_regular_grid(arch, FT)
+    model = TestModel_RegularRectGrid(arch, FT)
 
     Δt = FT(1.3e-6)
     Δx = FT(model.grid.Δx)
     u₀ = FT(1.2)
     CFL_by_hand = Δt * u₀ / Δx
 
-    model.velocities.u.data.parent .= u₀
+    set!(model, u=u₀)
     cfl = AdvectiveCFL(FT(Δt))
 
     return cfl(model) ≈ CFL_by_hand
 end
 
-function advective_cfl_diagnostic_is_correct_on_regular_grid(arch, FT)
-    model = TestModel_RegularRectGrid(arch, FT)
+function advective_cfl_diagnostic_is_correct_on_vertically_stretched_grid(arch, FT)
+    model = TestModel_VerticallyStrectedRectGrid(arch, FT)
 
     Δt = FT(1.3e-6)
     Δx = FT(model.grid.Δx)
     u₀ = FT(1.2)
     CFL_by_hand = Δt * u₀ / Δx
 
-    model.velocities.u.data.parent .= u₀
+    set!(model, u=u₀)
     cfl = AdvectiveCFL(FT(Δt))
 
     return cfl(model) ≈ CFL_by_hand
 end
 
-function diffusive_cfl_diagnostic_is_correct(arch, FT)
-    Δt = FT(1.3e-6)
-    Δx = FT(0.5)
-    ν = FT(1.2)
-    CFL_by_hand = Δt * ν / Δx^2
+function accurate_advective_cfl_on_regular_grid(arch, FT)
+    model = TestModel_RegularRectGrid(arch, FT)
 
-    model = TestModel_RegularRectGrid(arch, FT, ν, Δx)
-    cfl = DiffusiveCFL(FT(Δt))
+    Δt = FT(1.7)
+
+    Δx = model.grid.Δx
```

Comment:
If you only use each of these grid spacings once, would it be better to just use the full name where they are used?

Reply:
True. I think this change is part of https://github.com/CliMA/Oceananigans.jl/pull/1497 but I'll revise those tests before merging.